### PR TITLE
Refactor MTE-770 [v113] Update bitrise L10nScreenshot step

### PR DESCRIPTION
**Pull request by isabelrios**
PR to try to fix comments on issue #11744... Adding the comments on the issue here too:

- shiping_locales.txt file was used by the L10nScreenshots tests and also when a new locale was added / removed so that it was correctly imported. For the first part we are fine with removing the file, for the second part, we would need confirmation from iOS devs.

- we need to clone firefoxios-l10n repo to have the .xliff files so that the tests can run for each locale. When bootstrap runs without arguments, that repo is not deleted or cloned again. We prefer to clone it and not run bootstrap --importLocales becuse that imples the string import which takes a considerable amount of time and we don't need the very latest strings to run the tests.

- by running ./ios-l10n-scripts/import-locales-firefox.sh we were using the old tooling for the l10n, but only to get the list of xliff, in the tests we are not importing / exporting. Anyway by removing this part we are not using that repo in any place now

### CODE REVIEW

**Review by Delphine — COMMENTED** [review]
(sorry I don't think I can review this PR ;) thanks for the work on this)

**Comment by flodolo** [comment]
> * shiping_locales.txt file was used by the L10nScreenshots tests and also when a new locale was added / removed so that it was correctly imported. For the first part we are fine with removing the file, for the second part, we would need confirmation from iOS devs.

I don't think this needs another approval, given the build change has been done already (and approved by iOS devs)
https://github.com/mozilla-mobile/LocalizationTools/pull/2

> we need to clone firefoxios-l10n repo to have the .xliff files so that the tests can run for each locale. When bootstrap runs without arguments, that repo is not deleted or cloned again. We prefer to clone it and not run bootstrap --importLocales becuse that imples the string import which takes a considerable amount of time and we don't need the very latest strings to run the tests.

Why do you need a clone of the l10n repository at all? How are they used by the screenshots generation if they're not importe? I assumed `snapshot` would be using the strings that are already in the repository.

`l10n-screenshots.sh` has a check for the clone in questions, but what happens if you remove it?

```
if [ ! -d firefoxios-l10n ]; then
    echo "Did not find a firefox-ios-l10n checkout. Are you running this on a localized build?"
    exit 1
fi
```

File: bitrise.yml [hunk 1]
```diff
@@ -889,7 +889,7 @@ workflows:
             set -e
             set -x
 
-            # Import only the shipping locales (from shipping_locales.txt) onRelease
+            # Import  the locales from firefoxios-l10n
```
Comment (flodolo) [inline bitrise.yml:892]:
```suggestion
            # Import the locales from firefoxios-l10n
```

nit, pending the rest of the conversation

**Comment by isabelrios** [comment]
> Why do you need a clone of the l10n repository at all? How are they used by the screenshots generation if they're not importe? I assumed snapshot would be using the strings that are already in the repository.
> 
> l10n-screenshots.sh has a check for the clone in questions, but what happens if you remove it?
> 
> if [ ! -d firefoxios-l10n ]; then
>     echo "Did not find a firefox-ios-l10n checkout. Are you running this on a localized build?"
>     exit 1
> fi

@flodolo The tests run in a clean VM, if I don't clone the repo, the xliff files are not available and so the tests only run in English and we don't get the app localized to get the screenshots of the different screens on the available locales.

During the tests we don't import / export, we just run the tests. If the repo is not cloned, not sure how I can get the strings to be available for the tests... 

I have tried locally running the tests without that repo cloned and tests fail. 

I'm sorry, not sure what I'm missing or why we are not getting the same picture here... 
 



>

File: bitrise.yml [hunk 2]
```diff
@@ -959,23 +949,6 @@ workflows:
...
-            # Import only the shipping locales (from shipping_locales.txt) on Release
-            # builds. Import all locales on Beta and Fennec_Enterprise, except for pull
-            # requests.
-            git clone https://github.com/mozilla-mobile/ios-l10n-scripts.git ||exit 1
-            pip3 install --user virtualenv
-            cd /usr/local/bin
-            ln -s /Users/vagrant/Library/Python/3.10/bin/virtualenv .
-            cd -
-            virtualenv --version
-            ./ios-l10n-scripts/import-locales-firefox.sh --release
```
Comment (isabelrios) [inline bitrise.yml:978]: I can not test this on CI, but in theory, and based on your comment @flodolo we should be able to remove this part too.

This workflow is triggered after the L10nBuild workflow when the process to get the screenshots is run from Taskcluster. We will monitor this next time they run.

**Comment by isabelrios** [comment]
> > > During the tests we don't import / export, we just run the tests. If the repo is not cloned, not sure how I can get the strings to be available for the tests...
> > 
> > 
> > The strings are in repository, e.g. https://github.com/mozilla-mobile/firefox-ios/tree/main/Shared/it.lproj The build doesn't use XLIFF files, those are only used to import/export strings as an exchange format.
> > If you clone the repository and build, it should create a localized build, even without the l10n repository. Isn't that the case?
> > > I have tried locally running the tests without that repo cloned and tests fail.
> > 
> > 
> > What's the failure in the logs?
> 

@flodolo I was able to partially test removing that clone step. For the L10n build that works ([build](https://app.bitrise.io/build/1cda8cc7-789a-4ee6-b103-55079bc1a546?tab=artifacts) running it locale, it shows the correct language when running the tests), so I have removed it. It may be that I tried to remove it before having all the .lproj files in the project directory and I'm confused by that.

Once we run the complete process L10nBuild + L10nScreenshotsTests workflow I will let you know how this works (we can't run the second part from Bitrise locally, we need to trigger the Hook from Taskcluster) 
We are ready to land this and monitor the full process.

Thanks for your feedback and patience!

**Review by flodolo — APPROVED** [review]

File: l10n-screenshots.sh [hunk 3]
```diff
@@ -17,11 +17,6 @@ if [ -d l10n-screenshots ]; then
     exit 1
 fi
 
-if [ ! -d firefoxios-l10n ]; then
```
Comment (clarmso) [inline l10n-screenshots.sh:20]: 👍🏼

**Review by AaronMT — APPROVED** [review]

**Comment by nbhasin2** [comment]
@mergify backport release/v113

**Pull request merged**